ansible_runner/display_callback/events.py

```diff
@@ -134,40 +134,45 @@ def get(self):
         return ctx
 
     def get_begin_dict(self):
+        omit_event_data = os.getenv("RUNNER_OMIT_EVENTS", "False").lower() == "true"
+        include_only_failed_event_data = os.getenv("RUNNER_ONLY_FAILED_EVENTS", "False").lower() == "true"
         event_data = self.get()
-        # TODO: Need to rework these values
+        event = event_data.pop('event', None)
+        if not event:
+            event = 'verbose'
+            for key in ('debug', 'verbose', 'deprecated', 'warning', 'system_warning', 'error'):
+                if event_data.get(key, False):
+                    event = key
+                    break
+        event_dict = dict(event=event)
+        should_process_event_data = (include_only_failed_event_data and event in ('runner_on_failed', 'runner_on_async_failed', 'runner_on_item_failed')) \
```

Comment:
Should you future-proof this with `event.endswith('_failed')` ?

Reply:
It's a good question. I'm not sure I'm confident that would always match the desired pattern?

Reply:
🤷‍♂ I don't care a _lot_ about this.  I just know when `on_async_XYZ` became a thing, we had to update code in a bunch of places.